pkg/metrics/metrics.go

```diff
@@ -1329,24 +1322,19 @@ func NewLegacyMetrics() *LegacyMetrics {
 		}, []string{
 			LabelSourceCluster,
 			LabelSourceNodeName,
-			LabelTargetCluster,
-			LabelTargetNodeName,
-			LabelTargetNodeType,
 			LabelType,
+			LabelConnectivityStatus,
 		}),
 
-		NodeConnectivityLatency: metric.NewGaugeVec(metric.GaugeOpts{
+		NodeConnectivityLatency: metric.NewHistogramVec(metric.HistogramOpts{
```

Comment:
As well, can we update the [upgrade guide ](https://github.com/cilium/cilium/blob/main/Documentation/operations/upgrade.rst#changed-metrics) to mention this new metric (as well as a deprecation of the old one).

Reply:
hi @tommyp1ckles, I understand your concerns about being a breaking change. could you help me with understanding how this deprecation would work; specifically, where would I add a comment that this metric is to be removed in v1.18 / how do I add this as a new metric?

Reply:
+1 for updating `upgrade metrics`

> I do worry that this is a breaking change for anyone possibly using the existing metric (whether this is actually the case; I'm not sure).

I would consider making an exception for this metric - mostly because it's the most harmful metric we currently have. Let's discuss it at the next `Cilium Weekly Developer Meeting` maybe?